### PR TITLE
Expose component label via the KubeVirt CR

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:
Exposes relationship label "apps.kubevirt.io/component " to HCO via the KubeVirt CR,
set it's default value to be `kubevirt`.
Currently KubeVirt exposes relationship labels for "version" and "part-of" to HCO via the KubeVirt CR ,
The same needs to be done for "component". 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
 Designate the apps.kubevirt.io/component label for KubeVirt components.
```

### CODE REVIEW

**Comment by Barakmor1** [comment]
retest

File: pkg/virt-operator/resource/generate/install/strategy.go [hunk 1]
```diff
@@ -431,16 +431,24 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
 
 	var productName string
 	var productVersion string
+	var productComponent string
+
+	invalidLabelPatternErrorMassage := "invalid %s: labels must be 63 characters or less, begin and end with alphanumeric characters, and contain only dot, hyphen or dash"
```
Comment (dhiller) [inline pkg/virt-operator/resource/generate/install/strategy.go:436]: nit - spelling: s/Massage/Message/

File: pkg/virt-operator/resource/generate/install/strategy.go [hunk 2]
```diff
@@ -431,16 +431,24 @@ func GenerateCurrentInstallStrategy(config *operatorutil.KubeVirtDeploymentConfi
...
 
 	if operatorutil.IsValidLabel(config.GetProductName()) {
```
Comment (dhiller) [inline pkg/virt-operator/resource/generate/install/strategy.go:438]: nit: I see a repetition here, maybe we can extract this into a function and use that three times?

**Review by dhiller — APPROVED** [review]
I have some minor nits, otherwise looks good from my side. Since this is an api change, I'd ask @rmohr or @davidvossel to have a final look at this one.

/hold

**Review by dhiller — APPROVED** [review]
/approve

**Comment by dhiller** [comment]
/cc @rmohr @davidvossel

Could you PTAL? Thanks!

**Review by davidvossel — COMMENTED** [review]
/hold cancel
/approve

**Review by xpivarc — APPROVED** [review]

**Comment by Barakmor1** [comment]
retest

**Pull request merged**